src/main/java/fr/univartois/sonargo/coverage/CoverageSensor.java

```diff
@@ -49,79 +52,100 @@
 
 public class CoverageSensor implements Sensor {
 
-    private static final Logger LOGGER = Loggers.get(CoverageSensor.class);
+	private static final Logger LOGGER = Loggers.get(CoverageSensor.class);
 
-    @Override
-    public void describe(SensorDescriptor descriptor) {
-	descriptor.name("Go Coverage").onlyOnFileType(InputFile.Type.MAIN).onlyOnLanguage(GoLanguage.KEY);
-    }
-
-    public Stream<Path> createStream(SensorContext context) throws IOException {
-	final String fullPath = context.fileSystem().baseDir().getPath();
+	@Override
+	public void describe(SensorDescriptor descriptor) {
+		descriptor.name("Go Coverage").onlyOnFileType(InputFile.Type.MAIN).onlyOnLanguage(GoLanguage.KEY);
+	}
 
-	return Files.walk(Paths.get(fullPath))
-		.filter(p -> !p.getParent().toString().contains(".git") && !p.getParent().toString().contains(".sonar")
-			&& !p.getParent().toString().contains(".scannerwork")
-			&& !p.getFileName().toString().startsWith("."));
+	private List<String> getExcludedPath(SensorContext context) {
 
-    }
+		String globalExcludedPath = context.settings().getString(CoreProperties.GLOBAL_EXCLUSIONS_PROPERTY);
 
-    @Override
-    public void execute(SensorContext context) {
+		List<String> listExcludedPath = Arrays.asList(StringUtils.split(globalExcludedPath, ","));
 
-	try (Stream<Path> paths = createStream(context)) {
-	    paths.forEach(filePath -> {
-		if (Files.isDirectory(filePath)) {
-		    final String reportPath = context.settings().getString(GoProperties.COVERAGE_REPORT_PATH_KEY);
-		    final File f = new File(filePath + File.separator + reportPath);
-		    if (f.exists()) {
-			LOGGER.info("Analyse for " + f.getPath());
+		return listExcludedPath;
+	}
 
-			final CoverageParser coverParser = new CoverageParser(context);
+	private boolean isNotAnExcludedPath(Path p, SensorContext context) {
+		List<String> listExcludedPath = getExcludedPath(context);
 
-			try {
-			    coverParser.parse(f.getPath());
-			    save(context, coverParser.getCoveragePerFile());
-			} catch (ParserConfigurationException | SAXException | IOException e) {
-			    LOGGER.error("Exception: ", e);
+		for (String s : listExcludedPath) {
+			if (s.equals(p.toFile().getPath())) {
+				return false;
 			}
+		}
+		return true;
```

Comment:
Since this PR is revived, I wonder if the test should not be changed to p.toFile().getPath().startsWith(s) instead of the strict equality.

In that case, you may want to use streams to reduce the amount of code.